contracts/integrations/test/forwarder/forwarder_mainnet_test.ts

```diff
@@ -0,0 +1,201 @@
+            senderAddress: constants.NULL_ADDRESS,
+            feeRecipientAddress: constants.NULL_ADDRESS,
+            expirationTimeSeconds: new BigNumber(currentBlockTimestamp).plus(fifteenMinutesInSeconds),
+            salt: generatePseudoRandomSalt(),
+            makerAssetData,
+            takerAssetData: wethAssetData,
+            makerFeeAssetData: makerAssetData,
+            takerFeeAssetData: wethAssetData,
+            makerAssetAmount: constants.INITIAL_ERC20_BALANCE.dividedToIntegerBy(2),
+            takerAssetAmount: Web3Wrapper.toBaseUnitAmount(new BigNumber(2), 18),
+            makerFee: constants.ZERO_AMOUNT,
+            takerFee: constants.ZERO_AMOUNT,
+            ...orderParams,
+        };
+        const orderHashHex =
+            order.makerFeeAssetData === v2OrderId
+                ? (await exchangeV2.getOrderInfo(order).callAsync()).orderHash
+                : orderHashUtils.getOrderHashHex(order);
+        const signature = `0x${signingUtils
+            .signMessage(ethUtil.toBuffer(orderHashHex), makerPrivateKey, SignatureType.EthSign)
+            .toString('hex')}`;
+        return {
+            ...order,
+            signature,
+        };
+    }
+
+    describe('marketSellOrdersWithEth', () => {
+        it('should fill a single v2 order with no fees', async () => {
+            const order = await createOrderAsync({ makerFeeAssetData: v2OrderId });
+            const [wethSpentAmount, makerAssetAcquiredAmount] = await forwarder
+                .marketSellOrdersWithEth([order], [order.signature], [], [])
+                .callAsync({
+                    from: takerAddress,
+                    value: order.takerAssetAmount,
+                });
+            expect(wethSpentAmount).to.bignumber.eq(order.takerAssetAmount);
+            expect(makerAssetAcquiredAmount).to.bignumber.eq(order.makerAssetAmount);
+        });
+        it('should fill multiple v2 orders', async () => {
+            const orders = [
+                await createOrderAsync({ makerFeeAssetData: v2OrderId }),
+                await createOrderAsync({ makerFeeAssetData: v2OrderId }),
+            ];
+            const ethSellAmount = BigNumber.sum(
+                orders[0].takerAssetAmount,
+                orders[1].takerAssetAmount.dividedToIntegerBy(2),
+            );
+            const [wethSpentAmount, makerAssetAcquiredAmount] = await forwarder
+                .marketSellOrdersWithEth(orders, orders.map(o => o.signature), [], [])
+                .callAsync({
+                    from: takerAddress,
+                    value: ethSellAmount,
+                });
+            expect(wethSpentAmount).to.bignumber.eq(ethSellAmount);
+            expect(makerAssetAcquiredAmount).to.bignumber.eq(
+                BigNumber.sum(orders[0].makerAssetAmount, orders[1].makerAssetAmount.dividedToIntegerBy(2)),
+            );
+        });
+        it.skip('should fill multiple v2/v3 orders', async () => {
```

Comment:
Tried to play with this myself and couldn't figure out why it was reverting. The behavior is so bizarre. Are we sure the contract at `0x9ea309d4f96d6433c75a6b9c5a596cf19a87edb8` is the same one in this PR?

Reply:
I actually tried to get it to work by deploying a fresh Forwarder on the fork and running tests against that, same exact errors. I also selectively placed reverts to try to figure out what was going on and it appeared that`fillResults.protocolFeesPaid` was always 0 for v3 fills, despite getting past a `require(tx.gasprice != 0)` statement.